packages/types/src/codec/Enum.ts

```diff
@@ -64,7 +64,11 @@ function decodeFromJSON (registry: Registry, def: TypesDef, key: string, value?:
 
   assert(index !== -1, `Cannot map Enum JSON, unable to find '${key}' in ${keys.join(', ')}`);
 
-  return createFromValue(registry, def, index, value);
+  try {
+    return createFromValue(registry, def, index, value);
+  } catch (error) {
+    throw new Error(`Enum(${key}):: ${(error as Error).message}`);
+  }
```

Comment:
It is fine in general since it is included in https://github.com/polkadot-js/api/pull/1662, however not sure why it made in in here since this was reverted to be the types.